sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -153,6 +158,54 @@ impl Hyperdebug {
                             }
                         }
                     }
+                    "I2C" => {
+                        // We found the I2C forwarding USB interface (this one interface allows
+                        // multiplexing physical I2C ports.)
+                        let mut in_endpoint: Option<u8> = None;
+                        let mut out_endpoint: Option<u8> = None;
+                        for endpoint_desc in interface_desc.endpoint_descriptors() {
+                            if endpoint_desc.transfer_type() != rusb::TransferType::Bulk {
+                                continue;
+                            }
+                            match endpoint_desc.direction() {
+                                rusb::Direction::In => {
+                                    if let Some(_) = in_endpoint.replace(endpoint_desc.address()) {
+                                        return Err(Error::CommunicationError(
+                                            "Multiple I2C IN endpoints",
+                                        )
+                                        .into());
+                                    }
+                                }
+                                rusb::Direction::Out => {
+                                    if let Some(_) = out_endpoint.replace(endpoint_desc.address()) {
+                                        return Err(Error::CommunicationError(
+                                            "Multiple I2C OUT endpoints",
+                                        )
+                                        .into());
+                                    }
+                                }
+                            }
+                        }
+                        match (in_endpoint, out_endpoint) {
+                            (Some(in_endpoint), Some(out_endpoint)) => {
+                                if let Some(_) = i2c_interface.replace(BulkInterface {
+                                    interface: interface.number(),
+                                    in_endpoint,
+                                    out_endpoint,
+                                }) {
+                                    return Err(Error::CommunicationError(
+                                        "Multiple I2C interfaces",
+                                    )
+                                    .into());
+                                }
+                            }
+                            _ => {
+                                return Err(
```

Comment:
return is an expression, so I think you can just write `_ => return Err(...),`, although `bail!` might be more idiomatic.

Reply:
Done, here and elsewhere.